dts/bindings/serial/infineon,mtbhal-uart.yaml

```diff
@@ -0,0 +1,76 @@
+# Copyright (c) 2022, Cypress
```

Comment:
This is a copy with few changes from https://github.com/zephyrproject-rtos/zephyr/blob/b9e833366f1360ecbad8776d6e8587e180688e1b/dts/bindings/serial/cypress%2Cpsoc6-uart.yaml#L1-L34

You MUST keep copyrights

```diff
+# Copyright (c) 2018, Cypress
+# Copyright (c) 2020, ATL Electronics
# Copyright (c) 2022, Cypress
```

Reply:
done